python/lsst/daf/butler/butler.py

```diff
@@ -197,6 +197,21 @@ def __init__(self, config=None, collection=None, run=None):
             if self.run is None:
                 self.run = self.registry.makeRun(runCollection)
 
+    def __del__(self):
+        # Attempt to shut down any open resources when this object is
+        # garbage collected. Python does not guarantee that this method
+        # will be called, or that it will be called in the proper order
+        # so if possible the user should explicitly call shutdown. This
+        # exists only to make an attempt to shut things down as a last
+        # resort if at all possible.
+        self.shutdown()
+
+    def shutdown(self):
+        """This method should be called to properly close any resources the
+        butler may have open
```

Comment:
I'd like documentation to be extended a bit, in particular with information about how usable this instance is after calling `shutdown()` on it (or `close()` if you decide to rename it). I think it's probably safest to say that object is unusable, then just mention explicitly that  clients should not call any other methods after that.